src/handleQuery.ts

```diff
@@ -45,22 +52,22 @@ export async function makeUsageQueryJson<T = unknown>(ctx: Context, query: strin
                 message: 'No data found'
             };
         }
+
+        const total_results = result.rows_before_limit_at_least ?? 0;
         return {
             data: result.data,
-            status: 200 as ApiErrorResponse["status"],
-            meta: {
-                statistics: result.statistics ?? null,
-                rows: result.rows ?? 0,
-                rows_before_limit_at_least: result.rows_before_limit_at_least ?? 0,
-                request_time,
-                duration_ms: Number(new Date()) - Number(request_time),
-            }
+            statistics: result.statistics ?? {},
+            pagination: computePagination(page, limit, total_results),
+            results: result.rows ?? 0,
+            total_results,
+            request_time,
+            duration_ms: Number(new Date()) - Number(request_time),
         };
     } catch (err) {
         return {
             status: 500 as ApiErrorResponse["status"],
             code: "bad_database_response" as ApiErrorResponse["code"],
-            message: err
+            message: `${err}`
```

Comment:
Perhaps we might need extra handling

Isn't `err.message` the text response?

Not a big issue, this should still work

```js
message: `${err.message ?? err}`
```